mllib/src/main/scala/org/apache/spark/mllib/evaluation/RankingMetrics.scala

```diff
@@ -100,7 +100,7 @@ class RankingMetrics[T: ClassTag](predictionAndLabels: RDD[(Array[T], Array[T])]
           }
           i += 1
         }
-        precSum / labSet.size
+        precSum / n
```

Comment:
According to the definition linked to in wikipedia, the divisor should be the total number of relevant documents, which is labSet.size, or |D_i| above. Unless I'm missing an argument about why it has to be different, I think we should keep this and not change its answers, but update the formula in the docs.

Reply:
Yes, you are right. I was confused with the "Q" in the formula. I will update.
![screenshot from 2019-01-19 21-41-44](https://user-images.githubusercontent.com/23054875/51429276-1142a180-1c33-11e9-8459-d80137b99569.png)